sql: do not leak traces in lease renewal

tbg opened the pull request:
Found while investigating an unrelated problem via the /debug/requests endpoint.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/20044)
<!-- Reviewable:end -->

tbg committed:
sql: do not leak traces in lease renewal

Found while investigating an unrelated problem via the /debug/requests endpoint.

lgo reviewed (approved):
Welp, thanks for catching this. LGTM

tschottdorf commented:
TFTR!
